src/verify_multiple_aggregate_signatures.cc

```diff
@@ -0,0 +1,207 @@
+                .ThrowAsJavaScriptException();
+            return blst_ts::BLST_TS_ERROR::JS_ERROR_THROWN;
+        }
+        Napi::Object set = set_value.As<Napi::Object>();
+
+        Napi::Value msg_value = set.Get("message");
+        BLST_TS_FUNCTION_UNWRAP_UINT_8_ARRAY(msg_value, msg, "message")
+
+        sets.push_back(
+            {blst_ts::P1AffineGroup{},
+             blst_ts::P2AffineGroup{},
+             msg.Data(),
+             msg.ByteLength()});
+
+        blst_ts::BLST_TS_ERROR err =
+            unwrap_public_key(sets[i].pk_point, env, set.Get("publicKey"));
+        if (err != blst_ts::BLST_TS_ERROR::SUCCESS) {
+            return err;
+        }
+
+        err = unwrap_signature(sets[i].sig_point, env, set.Get("signature"));
+        if (err != blst_ts::BLST_TS_ERROR::SUCCESS) {
+            return err;
+        }
+    }
+
+    return blst_ts::BLST_TS_ERROR::SUCCESS;
+}
+
+/**
+ * @param[out]  {boolean} - Result of the verification
+ * @param[out]  {std::string} - Error message for invalid aggregate
+ * @param[in]   {BlstTsAddon *} - Addon module
+ * @param[in]   {blst::Paring *} - Pointer to paring for the verification
+ * @param[in]   {std::vector<SignatureSet>} - Sets to be added to pairing
+ */
+blst_ts::BLST_TS_ERROR verify_multiple_aggregate_signatures(
+    bool &result,
+    std::string &error_msg,
+    BlstTsAddon *module,
+    std::unique_ptr<blst::Pairing> &ctx,
+    std::vector<SignatureSet> &sets) {
+    for (uint32_t i = 0; i < sets.size(); i++) {
+        blst::byte rand[BLST_TS_RANDOM_BYTES_LENGTH];
+        if (!module->GetRandomNonZeroBytes(rand, BLST_TS_RANDOM_BYTES_LENGTH)) {
+            error_msg = "BLST_ERROR: Failed to generate random bytes";
+            return blst_ts::BLST_TS_ERROR::HAS_NATIVE_ERROR;
+        }
+
+        blst::BLST_ERROR err = ctx->mul_n_aggregate(
+            sets[i].pk_point.raw_point,
+            sets[i].sig_point.raw_point,
+            rand,
+            BLST_TS_RANDOM_BYTES_LENGTH * 8,
+            sets[i].msg,
+            sets[i].msg_len);
+        if (err != blst::BLST_ERROR::BLST_SUCCESS) {
+            error_msg = module->GetBlstErrorString(err) +
+                        ": Invalid batch aggregation at index "s +
+                        std::to_string(i);
+            return blst_ts::BLST_TS_ERROR::HAS_NATIVE_ERROR;
+        }
+    }
+    ctx->commit();
+    result = ctx->finalverify();
+    return blst_ts::BLST_TS_ERROR::SUCCESS;
+}
+
+Napi::Value VerifyMultipleAggregateSignatures(const Napi::CallbackInfo &info) {
+    BLST_TS_FUNCTION_PREAMBLE(info, env, module)
+    try {
+        bool result{false};
+        std::string error_msg{};
+        std::vector<SignatureSet> sets{};
+        blst_ts::BLST_TS_ERROR error =
+            prepare_verify_multiple_aggregate_signatures(sets, info);
+        if (error == blst_ts::BLST_TS_ERROR::SUCCESS) {
+            std::unique_ptr<blst::Pairing> ctx{
+                new blst::Pairing(true, module->dst)};
```

Comment:
In general, `make_unique` should be preferred. https://stackoverflow.com/questions/37514509/advantages-of-using-stdmake-unique-over-new-operator

Reply:
Updated.